x-pack/plugins/alerting/server/routes/rule/apis/list_types/transforms/transform_rule_types_response/v1.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isBoolean } from 'lodash/fp';
+import { RegistryAlertTypeWithAuth } from '../../../../../../authorization';
+import type { TypesRulesResponseBodyV1 } from '../../../../../../../common/routes/rule/apis/list_types';
+
+export const transformRuleTypesResponse = (
+  ruleTypes: Set<RegistryAlertTypeWithAuth>
+): TypesRulesResponseBodyV1 => {
+  return Array.from(ruleTypes).map((ruleType: RegistryAlertTypeWithAuth) => {
+    return {
+      ...(ruleType.actionGroups ? { action_groups: ruleType.actionGroups } : {}),
+      ...(ruleType.actionVariables ? { action_variables: ruleType.actionVariables } : {}),
+      ...(ruleType.alerts ? { alerts: ruleType.alerts } : {}),
```

Comment:
This one throws an error: Property 'alerts' does not exist on type 'RegistryAlertTypeWithAuth'. I'll add it to the type (see x-pack/plugins/alerting/server/rule_type_registry.ts changes), just let me know if you have more info/know a better option